python/lsst/cp/pipe/ptc.py

```diff
@@ -968,16 +965,23 @@ def errFunc(p, x, y):
                 msg = (f"\nSERIOUS: Not enough data points ({len(meanVecFinal)}) compared to the number of"
                        f"parameters of the PTC model({len(parsIniPtc)}). Setting {ampName} to BAD.")
                 self.log.warn(msg)
+                lenRawTimes = len(dataset.rawExpTimes)
+                lenNonLinPars = self.config.polynomialFitDegreeNonLinearity - 1
                 dataset.badAmps.append(ampName)
                 dataset.gain[ampName] = np.nan
                 dataset.gainErr[ampName] = np.nan
                 dataset.noise[ampName] = np.nan
                 dataset.noiseErr[ampName] = np.nan
-                dataset.nonLinearity[ampName] = np.nan
-                dataset.nonLinearityError[ampName] = np.nan
-                dataset.nonLinearityResiduals[ampName] = np.nan
-                dataset.fractionalNonLinearityResiduals[ampName] = np.nan
+                dataset.nonLinearity[ampName] = [np.nan for _ in range(lenRawTimes)]
+                dataset.nonLinearityError[ampName] = [np.nan for _ in range(lenRawTimes)]
```

Comment:
Shouldn't these two be the length of `lenNonLinPars` rather than the number of inputs?